src/api/credrank.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+
+import {CredGraph} from "../core/credrank/credGraph";
+import {
+  type WeightedGraph,
+  merge,
+  overrideWeights,
+} from "../core/weightedGraph";
+import {
+  ensureIdentityExists,
+  toBonusPolicy,
+  type DependenciesConfig,
+} from "../api/dependenciesConfig";
+import {Ledger} from "../core/ledger/ledger";
+import {computeBonusMinting, createBonusGraph} from "../core/bonusMinting";
+import {applyBudget, type Budget} from "../core/mintBudget";
+import {type WeightsT} from "../core/weights";
+import {contractions as identityContractions} from "../core/identity";
+import {credrank as computeCredrank} from "../core/credrank/compute";
+
+export type CredrankInput = {|
+  pluginGraphs: $ReadOnlyArray<WeightedGraph>,
+  ledger: Ledger,
+  dependencies: DependenciesConfig,
+  weightOverrides: WeightsT,
+  pluginsBudget: Budget | null,
+|};
+
+export type CredrankOutput = {|
+  credGraph: CredGraph,
+  updatedLedger: Ledger,
+  updatedDependencies: DependenciesConfig,
+|};
+
+/**
+  This is a primary SourceCred API that combines the given inputs into a single
+  WeightedGraph and then runs the Credrank algorithm on it to create a CredGraph
```

Comment:
nit: CredRank (caps) in prose. In code identifiers, we camel-case things
like `loadGithubRepo` and `runCredrank` because it’s more consistent
(otherwise you run into junk like `XMLHttpRequest`), but we use the
proper English spellings when spelling English.

Reply:
accepted